src/index.ts

```diff
@@ -3,8 +3,8 @@ const states = new WeakMap()
 class RemoteInputElement extends HTMLElement {
   constructor() {
     super()
-    const fetch = fetchResults.bind(null, this, true)
-    const state = {currentQuery: null, oninput: debounce(fetch), fetch, controller: null}
+    const fetch = (e: Event) => fetchResults.bind(null, this, true, e)
+    const state = {currentQuery: null, oninput: (e: Event) => debounce(fetch(e)), fetch, controller: null}
```

Comment:
I think this would be better:

```suggestion
    const state = {currentQuery: null, oninput: debounce((e: Event) => fetch(e)), fetch, controller: null}
```

Reply:
Good point!